akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -51,12 +51,15 @@ private[http] object HttpServerBluePrint {
     stack(settings, remoteAddress, log)
 
   def stack(settings: ServerSettings, remoteAddress: Option[InetSocketAddress], log: LoggingAdapter)(implicit mat: Materializer): Http.ServerLayer = {
-    userHandlerGuard(settings.pipeliningLimit) atop
-      requestPreparation(settings, remoteAddress) atop
-      controller(settings, log) atop
-      parsingRendering(settings, log) atop
-      websocketSupport(settings, log) atop
-      unwrapTls
+    val theStack =
+      userHandlerGuard(settings.pipeliningLimit) atop
+        requestPreparation(settings) atop
```

Comment:
or you zip it with a Source.maybe and send the address there?

-- 
Cheers,
√
On 2 Dec 2015 08:12, "Johannes Rudolph" <notifications@github.com> wrote:

> In
> akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala
> <https://github.com/akka/akka/pull/19060#discussion_r46382969>:
>
> > @@ -51,12 +51,15 @@ private[http] object HttpServerBluePrint {
> >      stack(settings, remoteAddress, log)
> >
> >    def stack(settings: ServerSettings, remoteAddress: Option[InetSocketAddress], log: LoggingAdapter)(implicit mat: Materializer): Http.ServerLayer = {
> > -    userHandlerGuard(settings.pipeliningLimit) atop
> > -      requestPreparation(settings, remoteAddress) atop
> > -      controller(settings, log) atop
> > -      parsingRendering(settings, log) atop
> > -      websocketSupport(settings, log) atop
> > -      unwrapTls
> > +    val theStack =
> > +      userHandlerGuard(settings.pipeliningLimit) atop
> > +        requestPreparation(settings) atop
>
> We could also turn this logic around and say the Attribute is another
> actual user API to get the peer address. It's not so much different from
> the hack the Remote-Address header is on the HTTP level to get metadata
> to the user.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19060/files#r46382969>.
>